modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/models/BrokerConfiguration.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License, 
+ * Version 2.0 (the "License"); you may not use this file except 
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.andes.configuration.models;
+
+import org.wso2.carbon.config.annotation.Configuration;
+import org.wso2.carbon.config.annotation.Element;
+
+/**
+ * Root configuration model of the Broker.
+ * This class provide access to all the configuration object representations of the broker.
+ */
+@Configuration(namespace = "wso2.carbon.broker",
+               description = "Broker configuration parameters")
+public class BrokerConfiguration {
+
+    @Element(description = "Coordination related configuration")
+    private CoordinationConfiguration coordination = new CoordinationConfiguration();
+
+    public CoordinationConfiguration getCoordination() {
```

Comment:
Do we need to provide getters and setters? Aren't they generated from `@Element` annotation?

Reply:
According to the sample in carbon-config, it is required.

Reply:
:+1:

Reply:
I checked back again. We only need getter methods. Setters not required. My bad. Thanks for pointing. https://github.com/wso2/carbon-config/blob/config-docs-plugin/samples/config-generator/src/main/java/org/wso2/carbon/config/samples/ParentConfiguration.java